mat-process-api.Tests/V1/UseCase/ProcessDataUseCaseTests.cs

```diff
@@ -83,5 +83,52 @@ public void check_gateway_returns_expected_response()
             Assert.AreEqual(response, result.ProcessData);
         }
 
+        #region Post Initial Process Document
+
+        //Ideally would have a test for whether the usecase calls factory method, unfortunatelly, there's no way to test that due to factory being a 'static' method, which can't be mocked.
+
+        [Test]
+        public void given_the_requestObject_when_executePost_usecase_method_is_called_then_the_gateway_is_called_with_factory_output() // this should be 2 tests really, one to see if factory gets called, and the other to see if gateway is called, but due to using static factory method this becomes impossible to separate.
+        {
+            //arrange
+            PostInitialProcessDocumentRequest requestObject = MatProcessDataHelper.CreatePostInitialProcessDocumentRequestObject();
+            MatProcessData domainObject = ProcessDataFactory.CreateProcessDataObject(requestObject);
+            mockMatGateway.Setup(g => g.PostInitialProcessDocument(It.IsAny<MatProcessData>())).Returns((MatProcessData dobj) => dobj.Id);
+            //act
+            processDataUseCase.ExecutePost(requestObject);
+
+            //assert
+            mockMatGateway.Verify(g => g.PostInitialProcessDocument(It.IsAny<MatProcessData>()), Times.Once);
+
+            // I need to include into assertion just enough object properties so that it would be demonstrated that all the data from the request is in a mapped-to object (probably 1 would be enough).
```

Comment:
Looks there's a better way. In the Verify bit you can check whether the mock object is called with the correct properties of the passed in object. Unfortunately checking the object itself does not work due to C# checking whether object references match instead of whether object contents match.

The call to factory in the UseCase gives ref1, while the call in the test (domainObject) gives ref2, hence test fails, if you don't check the contents of the properties themselves.